### PR TITLE
add GetLiveFiles and GetLiveFilesMetaData for BlobDB

### PR DESCRIPTION
**Pull request by miasantreble**

### CODE REVIEW

File: utilities/blob_db/blob_db_impl.cc [hunk 1]
```diff
@@ -948,6 +948,34 @@ Status BlobDBImpl::Write(const WriteOptions& opts, WriteBatch* updates) {
...
+  return Status::OK();
+}
+
+void BlobDBImpl::GetLiveFilesMetaData(std::vector<LiveFileMetaData>* metadata) {
+  db_->GetLiveFilesMetaData(metadata);
+  ReadLock rl(&mutex_);
+  for (auto bfile_pair : blob_files_) {
+    auto blob_file = bfile_pair.second;
+    LiveFileMetaData filemetadata;
+    filemetadata.size = blob_file->GetFileSize();
```
Comment (yiwu-arbug) [inline utilities/blob_db/blob_db_impl.cc]: Should fill `filemetadata.name` and `filemetadata.column_family_name` as well.

File: utilities/blob_db/blob_db_test.cc [hunk 2]
```diff
@@ -847,6 +847,26 @@ TEST_F(BlobDBTest, ColumnFamilyNotSupported) {
...
+  bdb_options.disable_background_tasks = true;
+  Open(bdb_options);
+  std::map<std::string, std::string> data;
+  for (size_t i = 0; i < 100; i++) {
+    PutRandom("key" + ToString(i), &rnd, &data);
+  }
+  auto *bdb_impl = static_cast<BlobDBImpl *>(blob_db_);
+  std::vector<LiveFileMetaData> metadata;
+  bdb_impl->GetLiveFilesMetaData(&metadata);
+  ASSERT_EQ(1U, metadata.size());
```
Comment (yiwu-arbug) [inline utilities/blob_db/blob_db_test.cc]: also verify the actual file names?

File: utilities/blob_db/blob_db_test.cc [hunk 3]
```diff
@@ -847,6 +847,26 @@ TEST_F(BlobDBTest, ColumnFamilyNotSupported) {
...
+  std::vector<std::string> livefile;
+  uint64_t mfs;
+  bdb_impl->GetLiveFiles(livefile, &mfs, false);
+  ASSERT_EQ(4U, livefile.size());
```
Comment (yiwu-arbug) [inline utilities/blob_db/blob_db_test.cc]: verify file names?

File: utilities/blob_db/blob_db_impl.cc [hunk 4]
```diff
@@ -948,6 +948,37 @@ Status BlobDBImpl::Write(const WriteOptions& opts, WriteBatch* updates) {
   return Status::OK();
 }
 
+Status BlobDBImpl::GetLiveFiles(std::vector<std::string>& ret,
+                                uint64_t* manifest_file_size,
+                                bool flush_memtable) {
+  ReadLock rl(&mutex_);
```
Comment (yiwu-arbug) [inline utilities/blob_db/blob_db_impl.cc]: Add comment on why holding mutex at the beginning?

Comment (miasantreble) [reply utilities/blob_db/blob_db_impl.cc]: to avoid updates to BaseDB during the for loop

Comment (yiwu-arbug) [reply utilities/blob_db/blob_db_impl.cc]: I mean, add an inline comment to explain?

**Pull request closed without merging**